### ISSUE TITLE
Compression format AVHuff not supported

### ISSUE THREAD

**Issue by edc71** [rtissera/libchdr#69]
AVHuff is the format used for laserdisks.. its used since v3 afaik

**Comment by rtissera** [comment]
@edc71 you're right.
I could add AVHuff support for sure but I have to find a way to not bloat too much libchdr which is meant to be easily embeddable in other projects with a small footprint.

**Comment by Sanaki** [comment]
Could always be optional support. If compiled with support, include requisite lib. That way it's available for projects that need it, but doesn't muss up those that don't. Though I suppose that does run contrary to recent efforts to make libchdr more uniform. Hrm.

**Comment by rtissera** [comment]
I agree with you @Sanaki meanwhile I had a look at AVHuff, hopefully it only depends on (possibly) flac which is already in through dr_flac and some huffman decoder so that's not a lot of code.
Worth having a look implementing it.
@edc71 what's the typical use case ? Daphne / Hypseus Singe ?

**Comment by Sanaki** [comment]
Seems like it, though I asked him to open this issue when the topic arose while looking at usage of libchdr with RomVault for data validation purposes. Obviously MAME would be a use case.

**Comment by edc71** [comment]
> I agree with you @Sanaki meanwhile I had a look at AVHuff, hopefully it only depends on (possibly) flac which is already in through dr_flac and some huffman decoder so that's not a lot of code. Worth having a look implementing it. @edc71 what's the typical use case ? Daphne / Hypseus Singe ?

Its actually for a tool which enters rominfo into a database. But hashing chd isnt enough, because the dats are using different hashes. So I needed something simple to include.

**Comment by rtissera** [comment]
Okay. I will start a dev branch to add it.
I also want to fix a bad warning (wrong free) and bump lzma so timing is good.

Will ping you and hypseus singe devs when done.

Envoyé de mon iPhone

**Comment by rtissera** [comment]
Closing as known limitation for now (preparing 0.3.0 release), feel free to re-open issue anytime after.

**Comment by rtissera** [comment]
Re opening with draft PR : https://github.com/rtissera/libchdr/pull/154

**Comment by rtissera** [comment]
@Sanaki @edc71 Draft PR #154 ready to merge, request for comments
